webapp/src/blocks.tsx

```diff
@@ -1089,7 +1089,10 @@ export class Editor extends toolboxeditor.ToolboxEditor {
                 return undefined;
             }
 
-            const blockId = this.compilationResult ? pxtblockly.findBlockIdByLine(this.compilationResult.sourceMap, { start: locInfo.line, length: locInfo.endLine - locInfo.line }) : undefined;
+            const blockId = locInfo.functionName === "<main>" /* Root JS function, therefore On Start block */ ?
```

Comment:
Would it be possible for a user to name their function "main" and cause problems here?

Reply:
Thankfully this isn't an issue, since a function named "main" will just have "main" as its function name, not "\<main\>". And since `<` and `>` are not allowed in js function names, a function named "\<main\>" just becomes "main" in the loc info, so still no issue there.

Tested as well, to confirm:
![image](https://github.com/user-attachments/assets/1b025f5c-fe11-4844-8455-b6517b197d9a)